core/src/conn.rs

```diff
@@ -36,60 +40,88 @@ impl MongoConnection {
         operation_timeout: Option<u32>,
         login_timeout: Option<u32>,
         type_mode: TypeMode,
+        mut runtime: Option<tokio::runtime::Runtime>,
     ) -> Result<Self> {
+        let runtime = runtime.take().unwrap_or_else(|| {
+            tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()
+                .unwrap()
+        });
```

Comment:
It's arguable we don't want to make this argument optional, but I like this as it makes using it directly int tests easier in the future.

Reply:
I'm fond of this